python/ray/data/_internal/execution/operators/actor_pool_map_operator.py

```diff
@@ -437,12 +477,29 @@ def current_size(self) -> int:
         return self.num_pending_actors() + self.num_running_actors()
 
     def num_running_actors(self) -> int:
-        return len(self._num_tasks_in_flight)
+        return len(self._running_actors)
+
+    def num_restarting_actors(self) -> int:
+        return sum(
+            1 if running_actor_state.is_restarting is True else 0
+            for running_actor_state in self._running_actors.values()
+        )
 
     def num_active_actors(self) -> int:
         return sum(
-            1 if num_tasks_in_flight > 0 else 0
-            for num_tasks_in_flight in self._num_tasks_in_flight.values()
+            1 if running_actor_state.num_tasks_in_flight > 0 else 0
+            for running_actor_state in self._running_actors.values()
+        )
+
+    def num_alive_actors(self) -> int:
+        return sum(
+            1
+            if (
+                running_actor_state.num_tasks_in_flight > 0
+                and running_actor_state.is_restarting is False
+            )
+            else 0
+            for running_actor_state in self._running_actors.values()
         )
```

Comment:
Does this mean we don't count an actor as alive if it doesn't have any tasks in flight?

Reply:
For scheduling tasks, we invoke pick_actors(). Earlier, it did not cover for the restarting case, but now it excludes actors restarting even with low in flight tasks.

Also resource accounting APIs current_processor_usage() and pending_processor_usage() account for restarting actors.